src/Test/QuickCheck/Gen.purs

```diff
@@ -160,9 +160,13 @@ runGen = runState
 evalGen :: forall a. Gen a -> GenState -> a
 evalGen = evalState
 
--- | Sample a random generator
-sample :: forall r a. Seed -> Size -> Gen a -> Array a
-sample seed sz g = evalGen (vectorOf sz g) { newSeed: seed, size: sz }
+-- | Sample a random generator, using the given seed and producing a sample
+-- | containing the given number of values.
+sample :: forall r a. Seed -> Int -> Gen a -> Array a
+sample seed n g = evalGen sampleGen { newSeed: seed, size: 0 }
+  where
+  sampleGen = traverse (flip resize g) sizes
```

Comment:
Perhaps something like

`sampleGen = traverse (flip resize g) (0 .. 20)`

To give singletons in arrays and also prevent extremely large sizes.